woocommerce/woocommerce-ios
# Hack Week: Site Health Status Checker under a feature flag

### Description
As part of the Hack Week, I did a sort of spike around the new Site Health Status Checker, implementing the first two screens (list + detail) and the view model for fetching all the data needed. In this first version, we fetch just the Orders list, Order detail, Products list, Product detail, Products Reviews List, Product Review detail. In the future, the idea is to integrate more endpoints.
Since this feature is intended to be used just for testing purposes, we want to fire just `GET` API calls and to use directly the Networking layer, because we don't want to store the data.
In addition, in this PR, I introduced async and await, since the functionality will involve having many nested callbacks, and we will have to wait for the results of a specific endpoint before executing another request (eg. we can fetch a single order only if we know the ID of that order, so this means that we should have a logic for fetching first the order list).
Currently, the data extracted are essentials: the response result (success or failure), the error, the time interval, and other small things.

### Testing instructions
1. Run the app in debug mode.
2. Open the Settings of the app.
3. You should see the new menu "Site Health Status Checker". Tap it.
4. You should be able to see the new functionality. Tap the button "Start Checking".
5. You should be able to open the result of every API request.

### Video
https://user-images.githubusercontent.com/495617/146577935-8c09c8c4-c36d-465d-96d6-97796b7237ea.mp4


---
